src/software/proto/primitive/primitive_msg_factory.h

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include "shared/proto/primitive.pb.h"
+#include "software/geom/angle.h"
+#include "software/geom/angular_velocity.h"
+#include "software/geom/point.h"
+#include "software/primitive/primitive.h"
+
+/**
+ * Enum for indicating the stop type for the Stop Primitive
+ */
+enum StopType
+{
+    BRAKE,
+    COAST
+};
+
+/**
+ * Create a Chip Primitive Message
+ *
+ * @param chip_origin The location where the chip will be taken
+ * @param chip_direction The orientation the Robot will chip at
+ * @param chip_distance_meters The distance between the starting location
+ * of the chip and the location of the first bounce
+ *
+ * @return Pointer to Chip Primitive Message
+ */
+std::unique_ptr<ChipPrimitiveMsg> createChipPrimitiveMsg(const Point &chip_origin,
```

Comment:
these should return PrimitiveNewMsg which is a `oneof` of ChipPrimitiveMsg, KickPrimitiveMsg, etc.

Reply:
Done.